vortex-serde/src/layouts/write/metadata_accumulators.rs

```diff
@@ -0,0 +1,243 @@
+//! Metadata accumulators track the per-chunk-of-a-column metadata, layout locations, and row counts.
+
+use std::sync::Arc;
+
+use vortex::array::StructArray;
+use vortex::stats::{ArrayStatistics as _, Stat};
+use vortex::validity::Validity;
+use vortex::{Array, IntoArray};
+use vortex_buffer::{Buffer, BufferString};
+use vortex_dtype::{match_each_native_ptype, DType, FieldName};
+use vortex_error::{VortexError, VortexResult};
+use vortex_scalar::Scalar;
+
+pub fn new_metadata_accumulator(dtype: &DType) -> Box<dyn MetadataAccumulator> {
+    match dtype {
+        DType::Null => Box::new(BasicAccumulator::new()),
```

Comment:
It occurs to me that we could probably also specialize all these to have/not-have null counts based on the nullability.

Reply:
I think we can defer this one for later. I’m not sure the null ability matters for performance here